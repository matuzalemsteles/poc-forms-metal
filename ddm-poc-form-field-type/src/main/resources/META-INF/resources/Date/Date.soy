{namespace Date}

{template .render}
    {@param? editable: bool}
    {@param? id: string}
    {@param? label: string}
    {@param? placeholder: string}
    {@param? required: bool}
    {@param? showLabel: bool}
    {@param? spritemap: string}
    {@param? value: string}

    {call FieldBase.render}
        {param contentRenderer kind="html"}
            {call .content}
                {param editable: $editable /}
                {param placeholder: $placeholder /}
                {param spritemap: $spritemap /}
                {param value: $value /}
            {/call}
        {/param}

        {param id: $id /}
        {param label: $label /}
        {param required: $required /}
        {param showLabel: $showLabel /}
        {param spritemap: $spritemap /}
    {/call}
{/template}

{template .content}
    {@param? editable: bool}
    {@param? placeholder: string}
    {@param? spritemap: string}
    {@param? value: string}

    {let $attributes kind="attributes"}
        class="form-control"

        {if $placeholder}
            placeholder="{$placeholder}"
        {/if}

        {if $value}
            value="{$value}"
        {/if}

        type="text"

        {if not $editable}
            disabled
        {/if}
    {/let}

    <div class="input-group">
        <div class="input-group-item input-group-prepend">
            <input {$attributes}>
        </div>

        {if $spritemap}
            <div class="input-group-append input-group-item input-group-item-shrink">
                <span class="input-group-text">
                    {call ClayIcon.render}
                        {param spritemap: $spritemap /}
                        {param symbol: 'calendar' /}
                    {/call}
                </span>
            </div>
        {/if}
    </div>
{/template}