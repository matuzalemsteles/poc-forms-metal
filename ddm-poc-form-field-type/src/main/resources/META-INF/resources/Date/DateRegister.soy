{namespace DateRegister}

{deltemplate LayoutRenderer.RegisterFieldType variant="'date'"}
    {call Date.render data="all" /}
{/deltemplate}