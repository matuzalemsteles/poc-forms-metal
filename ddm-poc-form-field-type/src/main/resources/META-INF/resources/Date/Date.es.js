import 'clay-icon';
import {Config} from 'metal-state';
import Component from 'metal-component';
import FieldBase from 'ddm-poc-form-field-type/FieldBase/index.js';
import Soy from 'metal-soy';

import templates from './Date.soy.js';
import DateRegister from './DateRegister.soy.js';

/**
 * Date.
 * @extends Component
 */
class Date extends Component {
    static STATE = {
        /**
         * @default false
         * @instance
         * @memberof Date
         * @type {?bool}
         */
        editable: Config.bool().value(false),

        /**
         * @default undefined
         * @instance
         * @memberof Date
         * @type {?(string|undefined)}
         */
        helpText: Config.string(),

        /**
         * @default undefined
         * @instance
         * @memberof Date
         * @type {?(string|undefined)}
         */
        id: Config.string(),

        /**
         * @default undefined
         * @instance
         * @memberof Date
         * @type {?(string|undefined)}
         */
        label: Config.string(),

        /**
         * @default __/__/____
         * @instance
         * @memberof Date
         * @type {?string}
         */
        placeholder: Config.string().value('__/__/____'),

        /**
         * @default false
         * @instance
         * @memberof Date
         * @type {?bool}
         */
        required: Config.bool().value(false),

        /**
         * @default true
         * @instance
         * @memberof Date
         * @type {?bool}
         */
        showLabel: Config.bool().value(true),

        /**
         * @default undefined
         * @instance
         * @memberof Date
         * @type {?(string|undefined)}
         */
        spritemap: Config.string(),

        /**
         * @default undefined
         * @instance
         * @memberof Date
         * @type {?(string|undefined)}
         */
        value: Config.string(),
    }
}

Soy.register(Date, templates);

export default Date;