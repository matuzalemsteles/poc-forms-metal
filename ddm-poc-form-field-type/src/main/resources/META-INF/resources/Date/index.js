import Date from './Date.es';

export default Date;