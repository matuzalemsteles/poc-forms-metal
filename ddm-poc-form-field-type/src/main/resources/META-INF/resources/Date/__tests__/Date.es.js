import Date from '../Date.es';

let component;
let spritemap = 'icons.svg';

describe('Field Date', () => {
    afterEach(() => {
        if (component) {
			component.dispose();
        }
    });

    it('should be not edidable', () => {
        component = new Date({
            editable: false,
            spritemap: spritemap,
        });

        expect(component).toMatchSnapshot();
    });

    it('should have a helpText', () => {
        component = new Date({
            helpText: 'Type something',
            spritemap: spritemap,
        });

        expect(component).toMatchSnapshot();
    });

    it('should have an id', () => {
        component = new Date({
            id: 'ID',
            spritemap: spritemap,
        });

        expect(component).toMatchSnapshot();
    });

    it('should have a label', () => {
        component = new Date({
            label: 'label',
            spritemap: spritemap,
        });

        expect(component).toMatchSnapshot();
    });

    it('should have a placeholder', () => {
        component = new Date({
            placeholder: '__/__/____',
            spritemap: spritemap,
        });

        expect(component).toMatchSnapshot();
    });

    it('should not be required', () => {
        component = new Date({
            required: false,
            spritemap: spritemap,
        });

        expect(component).toMatchSnapshot();
    });

    it('should render Label if showLabel is true', () => {
        component = new Date({
            label: 'text',
            showLabel: true,
            spritemap: spritemap,
        });

        expect(component).toMatchSnapshot();
    });

    it('should have a spritemap', () => {
        component = new Date({
            spritemap: spritemap
        });

        expect(component).toMatchSnapshot();
    });

    it('should have a value', () => {
        component = new Date({
            value: 'value',
            spritemap: spritemap,
        });

        expect(component).toMatchSnapshot();
    });
});