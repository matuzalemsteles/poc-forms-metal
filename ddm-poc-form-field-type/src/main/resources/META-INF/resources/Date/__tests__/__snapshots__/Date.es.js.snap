// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field Date should be not edidable 1`] = `
<div class="form-group">
  <label></label>
  <div class="input-group">
    <div class="input-group-item input-group-prepend">
      <input class="form-control" placeholder="__/__/____" type="text" disabled="">
    </div>
    <div class="input-group-append input-group-item input-group-item-shrink">
      <span class="input-group-text">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-calendar" focusable="false">
          <title>calendar</title>
          <use xlink:href="icons.svg#calendar"></use>
        </svg>
      </span>
    </div>
  </div>
  <span class="form-text"></span>
</div>
`;

exports[`Field Date should have a helpText 1`] = `
<div class="form-group">
  <label></label>
  <div class="input-group">
    <div class="input-group-item input-group-prepend">
      <input class="form-control" placeholder="__/__/____" type="text" disabled="">
    </div>
    <div class="input-group-append input-group-item input-group-item-shrink">
      <span class="input-group-text">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-calendar" focusable="false">
          <title>calendar</title>
          <use xlink:href="icons.svg#calendar"></use>
        </svg>
      </span>
    </div>
  </div>
  <span class="form-text"></span>
</div>
`;

exports[`Field Date should have a label 1`] = `
<div class="form-group">
  <label>label </label>
  <div class="input-group">
    <div class="input-group-item input-group-prepend">
      <input class="form-control" placeholder="__/__/____" type="text" disabled="">
    </div>
    <div class="input-group-append input-group-item input-group-item-shrink">
      <span class="input-group-text">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-calendar" focusable="false">
          <title>calendar</title>
          <use xlink:href="icons.svg#calendar"></use>
        </svg>
      </span>
    </div>
  </div>
  <span class="form-text"></span>
</div>
`;

exports[`Field Date should have a placeholder 1`] = `
<div class="form-group">
  <label></label>
  <div class="input-group">
    <div class="input-group-item input-group-prepend">
      <input class="form-control" placeholder="__/__/____" type="text" disabled="">
    </div>
    <div class="input-group-append input-group-item input-group-item-shrink">
      <span class="input-group-text">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-calendar" focusable="false">
          <title>calendar</title>
          <use xlink:href="icons.svg#calendar"></use>
        </svg>
      </span>
    </div>
  </div>
  <span class="form-text"></span>
</div>
`;

exports[`Field Date should have a spritemap 1`] = `
<div class="form-group">
  <label></label>
  <div class="input-group">
    <div class="input-group-item input-group-prepend">
      <input class="form-control" placeholder="__/__/____" type="text" disabled="">
    </div>
    <div class="input-group-append input-group-item input-group-item-shrink">
      <span class="input-group-text">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-calendar" focusable="false">
          <title>calendar</title>
          <use xlink:href="icons.svg#calendar"></use>
        </svg>
      </span>
    </div>
  </div>
  <span class="form-text"></span>
</div>
`;

exports[`Field Date should have a value 1`] = `
<div class="form-group">
  <label></label>
  <div class="input-group">
    <div class="input-group-item input-group-prepend">
      <input class="form-control" placeholder="__/__/____" value="value" type="text" disabled="">
    </div>
    <div class="input-group-append input-group-item input-group-item-shrink">
      <span class="input-group-text">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-calendar" focusable="false">
          <title>calendar</title>
          <use xlink:href="icons.svg#calendar"></use>
        </svg>
      </span>
    </div>
  </div>
  <span class="form-text"></span>
</div>
`;

exports[`Field Date should have an id 1`] = `
<div class="form-group">
  <label for="ID"></label>
  <div class="input-group">
    <div class="input-group-item input-group-prepend">
      <input class="form-control" placeholder="__/__/____" type="text" disabled="">
    </div>
    <div class="input-group-append input-group-item input-group-item-shrink">
      <span class="input-group-text">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-calendar" focusable="false">
          <title>calendar</title>
          <use xlink:href="icons.svg#calendar"></use>
        </svg>
      </span>
    </div>
  </div>
  <span class="form-text"></span>
</div>
`;

exports[`Field Date should not be required 1`] = `
<div class="form-group">
  <label></label>
  <div class="input-group">
    <div class="input-group-item input-group-prepend">
      <input class="form-control" placeholder="__/__/____" type="text" disabled="">
    </div>
    <div class="input-group-append input-group-item input-group-item-shrink">
      <span class="input-group-text">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-calendar" focusable="false">
          <title>calendar</title>
          <use xlink:href="icons.svg#calendar"></use>
        </svg>
      </span>
    </div>
  </div>
  <span class="form-text"></span>
</div>
`;

exports[`Field Date should render Label if showLabel is true 1`] = `
<div class="form-group">
  <label>text </label>
  <div class="input-group">
    <div class="input-group-item input-group-prepend">
      <input class="form-control" placeholder="__/__/____" type="text" disabled="">
    </div>
    <div class="input-group-append input-group-item input-group-item-shrink">
      <span class="input-group-text">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-calendar" focusable="false">
          <title>calendar</title>
          <use xlink:href="icons.svg#calendar"></use>
        </svg>
      </span>
    </div>
  </div>
  <span class="form-text"></span>
</div>
`;
